core/src/wsv.rs

```diff
@@ -430,17 +430,31 @@ impl WorldStateView {
     /// Example: account events before domain events.
     ///
     /// # Errors
-    /// Fails if `f` fails
-    ///
-    /// # Panics
-    /// (Rare) Panics if can't lock `self.events` for writing
-    #[allow(clippy::unwrap_in_result, clippy::expect_used)]
+    /// Forward errors from [`Self::modify_world_multiple_events`]
     pub fn modify_world(
         &self,
         f: impl FnOnce(&World) -> Result<WorldEvent, Error>,
     ) -> Result<(), Error> {
-        let world_event = f(&self.world)?;
-        let data_events: SmallVec<[DataEvent; 3]> = world_event.into();
+        self.modify_world_multiple_events(move |world| f(world).map(std::iter::once))
+    }
+
+    /// Get `World` and pass it to closure to modify it
+    ///
+    /// Produces events in the `WSV` that are produced by `f` during execution.
```

Comment:
"produces events ... that are produced" -- this should be rephrased

maybe like this (if I understood you correctly): "The function takes events from `WSV` that were produced during the execution of `f` and modifies them."?

Reply:
> "produces events ... that are produced" -- this should be rephrased

Yeah, this sound confusing.

I would slightly change you suggestion: "The function put events produced during `f` execution into `event_buffer`"